### DISCUSSION TITLE
v1.1.12

### DISCUSSION THREAD

**Discussion by coreybutler** [coreybutler/nvm-windows#1066] [General]
## What's Changed
* Fix symlink detect if it points to nothing by @Tester798 in https://github.com/coreybutler/nvm-windows/pull/963
* Update README.md by @coreybutler in https://github.com/coreybutler/nvm-windows/pull/1005
* Change Winget Releaser job to `ubuntu-latest` by @sitiom in https://github.com/coreybutler/nvm-windows/pull/1016
* Update README.md by @coreybutler in https://github.com/coreybutler/nvm-windows/pull/1034
* Modified folder detection code against #1031 by @Humbunklung in https://github.com/coreybutler/nvm-windows/pull/1032
* Update bug_report.yml by @coreybutler in https://github.com/coreybutler/nvm-windows/pull/1040
* Update bug_report.yml by @coreybutler in https://github.com/coreybutler/nvm-windows/pull/1041
* Update bug_report.yml by @coreybutler in https://github.com/coreybutler/nvm-windows/pull/1042
* Fix small typo in README.md by @acdvorak in https://github.com/coreybutler/nvm-windows/pull/1048
* Avoid unnecessary byte/string conversion by @Juneezee in https://github.com/coreybutler/nvm-windows/pull/1049
* fix issue #932 by @nineninesevenfour in https://github.com/coreybutler/nvm-windows/pull/965
* allow to run build.bat on paths with spaces by @nineninesevenfour in https://github.com/coreybutler/nvm-windows/pull/967

## New Contributors
* @Tester798 made their first contribution in https://github.com/coreybutler/nvm-windows/pull/963
* @coreybutler made their first contribution in https://github.com/coreybutler/nvm-windows/pull/1005
* @Humbunklung made their first contribution in https://github.com/coreybutler/nvm-windows/pull/1032
* @acdvorak made their first contribution in https://github.com/coreybutler/nvm-windows/pull/1048
* @Juneezee made their first contribution in https://github.com/coreybutler/nvm-windows/pull/1049
* @nineninesevenfour made their first contribution in https://github.com/coreybutler/nvm-windows/pull/965

**Full Changelog**: https://github.com/coreybutler/nvm-windows/compare/1.1.11...1.1.12

<hr /><em>This discussion was created from the release <a href='https://github.com/coreybutler/nvm-windows/releases/tag/1.1.12'>v1.1.12</a>.</em>

**Comment by sitiom** [comment]
@coreybutler Winget releaser workflow failed due to the `nvm-setup.exe` installer not available in the releases: https://github.com/coreybutler/nvm-windows/actions/runs/6962150119/job/18945262798.

This seems to be a case of https://github.com/coreybutler/nvm-windows/pull/885#issuecomment-1506295022 again. Can you please upload it and rerun the workflow again? Thanks!

**Reply by zhuoNy** [reply]
![nvm](https://github.com/coreybutler/nvm-windows/assets/60495408/5774e452-4604-4c54-b3ef-84d309da2306)
Why not support other terminal, just like Git bash. thank you.(v1.1.12)

**Reply by coreybutler** [reply]
`nvm-setup.exe` has been added.

@zhuoNy this message usually shows up when you try to run NVM4W by double-clicking on it (i.e. interactively, like a GUI program). Using git bash isn't excluded, but we don't support it because it is a non-standard shell. There are lots of non-standard shells and we can't support them all... especially those which change how paths are defined in Windows (think Cygwin with unix paths).

**Reply by sitiom** [reply]
> `nvm-setup.exe` has been added.

Can you rerun the workflow again?

**Reply by coreybutler** [reply]
@sitiom I have, but the action still fails internally (fails to create an upstream branch). I thought this was fixed last time, but it doesn't appear to be. I don't have time to troubleshoot this right now (US holiday weekend).

**Reply by sitiom** [reply]
No worries!

```
  Failed to create branch from upstream default branch
  node:internal/errors:866
    const err = new Error(message);
```

~My guess is that either the token you provided has expired or does not have the proper `public_repo` scope.~ https://github.com/coreybutler/nvm-windows/issues/1091#issuecomment-1935201802

**Reply by ezhupa99** [reply]
I just get the same issue as the first commenter when I ran the **nvm-update.exe** but then the version got updated

![image](https://github.com/coreybutler/nvm-windows/assets/34006441/cc5da728-d3e0-420f-9fb2-25c9d9861911)

**Reply by coreybutler** [reply]
@ezhupa99 there is an open issue for this. For others coming across this via Google, just reinstall with the main installer.

**Reply by Ernest3688** [reply]
what should i type on my terminal when i want to use the latest nvm

**Reply by coreybutler** [reply]
There is no command to update to the latest _nvm_ from the terminal. We don't have an auto-update service for this project. That is something we're working on for the commercial version of [Runtime](https://github.com/coreybutler/nvm-windows/wiki/Runtime).

If you actually meant "what should I type on my terminal when I want to use the latest _node_", the answer is `nvm use latest`. That will use the latest version you have installed. If you want to install the latest version, run `nvm install latest`.

**Comment by frremu** [comment]
Please add .exe resource in Assets. 

Freundliche Grüße.

**Reply by coreybutler** [reply]
<img width="895" alt="image" src="https://github.com/coreybutler/nvm-windows/assets/770982/90e4d85e-0366-44c7-a66d-0d7e4d72d275">

**Reply by rchak007** [reply]
@coreybutler What are the steps to install?? do i download all assets?? its confusing. or just the zip. Do i get all the files.??

**Reply by coreybutler** [reply]
@rchak007 just download nvm-setup.exe. Then run the file. That's it. The installation wizard walks you through the rest. The other files aren't necessary for new installations. 

The nvm-noinstall package is for people who want to install manually without the wizard (there is a guide in the wiki). The nvm-update package is for people updating to a new version of NVM for Windows... but all it does is reinstall with fewer prompts. All of the text files are just checksums... the majority of people won't need this since the purpose of checksums is just to validate the software is legit. Since we code-sign our executables, Windows itself verifies our code... so the checksums really aren't necessary.

**Reply by rchak007** [reply]
ok thx. i think i downloaded the zip but was able to run it and install.. thanks a lot!! was able to work it

**Reply by 20Lov** [reply]
我还是不会安装，下载了nvm-setup.exe之后要运行nvm.exe吗还是怎么整
希望能给一个安装教程，纯新手

**Comment by u561910** [comment]
What would really be great if you could create an installer that installs to the system instead of to a user profile. Currently our Devs are unable to use your application because it is blocked by security policies. As you probably know, installing EXEs and DLLS, etc.....can make it much easier for malicious code to launch from user profiles. User profile installers can be launched without admin rights

**Reply by coreybutler** [reply]
@u561910 I'm not sure I follow the logic here. Running the code at a system level grants all users access. It does not change the fact that Windows itself requires elevated permissions to create symlinks and access directories like `C:\Program Files` (if you choose to install it this way... you can use other directories). From what I can see, a system-level installation would increase the vulnerability surface area, not reduce it. It also makes it difficult to identify compromised accounts.

There are thousands of organizations deploying this to enterprise developers via profiles, for the reasons stated above. If you need a different installer to meet your firm's specific policies, there is a guide for manual installation in the wiki. Some companies use this to create their own installers.

Some firms need an MSI installer instead of the EXE, which helps control how the app is installed in private environments. If this is something you want, you'll need to wait for [Runtime](https://github.com/coreybutler/nvm-windows/wiki/Runtime). Keep in mind, enterprise features will be part of a commercial offering.

That said, I'm not trying to dismiss your concerns... I just don't understand where the problem is.

**Comment by v10o** [comment]
I just want to thank you for all you do. What you have developed is a life changing experience.

**Comment by samueloshio** [comment]
nvm-setup.exe v1.1.12 would not download. Please help check

**Reply by coreybutler** [reply]
The download works correctly.

<img width="308" alt="image" src="https://github.com/coreybutler/nvm-windows/assets/770982/ecbc2a09-bc7b-47b3-bf82-ae6afc8f2a68">

Make sure your system is allowed to download `.exe` files. Tools like Windows Defender are often configured to block direct downloads of executables. That's why we offer a zip file version too.

**Reply by Tebza98** [reply]
Do i have to download all the assets?

**Reply by coreybutler** [reply]
@Tebza98 no, you just need one. Most people choose `nvm-setup.exe` (or `nvm-setup.zip` if your firewall blocks .exe downloads), which provides the complete installer/uninstaller. If you need to make your own installer, choose one of the `noinstall` options, which provides everything except the installer (wiki has instructions for manual install).

**Comment by protypangel** [comment]
It's me or the nvm-setup.exe has a problem with the variable systeme ?
That create the variable "NVM_HOME" when the install.cmd need the variable system "NVM_PATH"

**Reply by coreybutler** [reply]
@protypangel - `NVM_HOME` is the correct variable used by NVM for Windows. `NVM_PATH` is used by [nvm-sh](https://github.com/nvm-sh/nvm). Maybe you're thinking of that project?

**Comment by RPWPA** [comment]
Why does virustotal say it has got malicious content in the setup.exe file?
https://www.virustotal.com/gui/url/870689a1d8da4b144b6599d2f0653deae9e2ec65d7cb1c4d18042f83d84b1eaa?nocache=1

**Reply by coreybutler** [reply]
@RPWPA First, two vendors out of 92 say there is malicious code, and one says suspicious. This happens every now and again. The most common reason is the antivirus vendors sometimes use unsophisticated ML to determine threat levels. It's not uncommon to see _ALL_ Go apps falsely flagged.

I've worked with several major enterprises and governments who've provided Crowdstrike Falcon reports with the same flags. In every case, these antivirus tools have falsely assumed `nvm-setup.exe` is `nvm.exe`, and every flag is against `nvm-setup.exe`. Their logic is a program shouldn't be able to install anything, which is reasonable... unless you're treating the installer as a program!

Some of these tools also flag references to WMI/WMIC and RPC calls, which is done internally by Windows when an HTTP request is made (to download Node.js/npm).

The bottom line is there are false positives. I try to work with the AV vendors on these, but so many of them pop up and disappear before I can accomplish anything with them.

The guidance I provide to enterprise users is pretty basic... there are no credible threats. However; if you feel better not using the installer, the [manual installation process](https://github.com/coreybutler/nvm-windows/wiki#manual-installation) provides an alternative that doesn't use nvm-setup.exe at all.

**Reply by RPWPA** [reply]
Thanks for the help @coreybutler.

**Comment by vprbo** [comment]
Hi, thanks for the release! Is there a nvm-setup.exe command line to override the default nodejs symlink path?

**Reply by coreybutler** [reply]
The GUI installer provides an option for this.

**Reply by vprbo** [reply]
Yes I saw that, I was wondering if there is a command line for it as I couldn't find a parameter to set the path, such as "nvm-setup.exe /SYMLINK=C:\Program Files\Node". Am I missing something?

**Reply by coreybutler** [reply]
There aren't any flags/arguments for the setup executable. It is possible to create a custom installer if you need to though. The wiki has a manual installation guide that some folks use to create custom company-wide installers.

**Comment by Py-Ec** [comment]
what should i type on my terminal when i want to use the latest nvm

**Reply by coreybutler** [reply]
<img width="975" alt="image" src="https://github.com/coreybutler/nvm-windows/assets/770982/d86e322a-7145-4659-abc8-35ff5021c408">

To clarify:

`newest` is the newest version installed on your computer. `latest` is the absolute latest release available. `lts` is the newest long term support release. The `latest` and `lts` are more commonly used with `nvm install`, but they are available for `nvm use` for those who like to know when they are no longer using the latest/LTS version (because NVM4W will tell you it isn't installed once a newer version is available for download).

**Comment by winter0yu** [comment]
Why does the checksum you provided show a verification error when checked with the 7zip CRC tool?

**Reply by coreybutler** [reply]
Probably because it's an MD5 checksum, not a CRC checksum. You can see how it is created [here](https://github.com/coreybutler/nvm-windows/blob/master/build.bat#L94).

**Comment by ThirumuruganG** [comment]
![nvm](https://github.com/user-attachments/assets/cd0a185d-32ff-4729-9277-bca15ead7457)
Hello butler,
I have a que, Can you please advise how i can process after download the nvm-noinstall ?
Here i waiting for your response.

**Reply by coreybutler** [reply]
@ThirumuruganG that's up to you. That question is just asking where you want to extract the files to (that's just Windows asking you that). The no-install file is for people who want to [manually install the application](https://github.com/coreybutler/nvm-windows/wiki#installation). Otherwise you should use the nvm-setup file, which provides a standard installer.

**Reply by ThirumuruganG** [reply]
Okay will do.
Thank you!

**Comment by ThirumuruganG** [comment]
![image](https://github.com/user-attachments/assets/2313172d-3af2-4677-bc45-f1bb4dc6979a)
Here i attached the screenshot of the error which was shown in our system while we enter the npm in VScode,
please advise me how to fix the issue.

**Reply by coreybutler** [reply]
I think this error is self-evident: `npm.ps1 cannot be loaded because running scripts is disabled on this system` That is not an NVM for Windows problem. It is a system-level permissions error. You'll need to talk to your system administrators. See the links provided in your error messages for details.

**Reply by ThirumuruganG** [reply]
Okay, Thanks!

**Comment by bard83** [comment]
Hi,
installed nvm using the nvm-noinstall version. The env variables have been defined as described but when I run `nvm debug` an interesting error has returned: `could not be found or does not exist. Exiting.`

Below the complete output:

```powershell
PS C:\Users\xXx> nvm debug
 could not be found or does not exist. Exiting.
xXx is not using admin or elevated rights.

Windows PowerShell
Windows Version: 10.0 (Build 26100)

Windows Developer Mode: UNKNOWN (user cannot read registry)

NVM4W Version:      1.1.12
NVM4W Path:         C:\Users\xXx\AppData\Roaming\nvm\nvm.exe
NVM4W Settings:     C:\Users\xXx\Roaming\nvm\settings.txt
NVM_HOME:           C:\Users\xXx\AppData\Roaming\nvm
NVM_SYMLINK:        C:\Users\xXx\AppData\Roaming\nodejs
Node Installations:

Total Node.js Versions: 0
Active Node.js Version: none
(run "nvm use <version>" to activate a version)
NVM_SYMLINK does not exist yet. This is auto-created when "nvm use" is run.

IPv6 is enabled. This can slow downloads significantly.

No problems detected.

Find help at https://github.com/coreybutler/nvm-windows/wiki/Common-Issues
```

The error blocks the node installation:
```powershell
PS C:\Users\xXx> nvm install 22.12.0
 could not be found or does not exist. Exiting.
Downloading node.js version 22.12.0 (64-bit)...
Error while creating \v22.12.0\node.zip - open \v22.12.0\node.zip: The system cannot find the path specified.
Could not download node.js v22.12.0 64-bit executable.
```
